parsing/lexer.mll

```diff
@@ -627,6 +662,10 @@ and string = parse
   | '\\' 'x' ['0'-'9' 'a'-'f' 'A'-'F'] ['0'-'9' 'a'-'f' 'A'-'F']
       { store_escaped_char lexbuf (char_for_hexadecimal_code lexbuf 2);
          string lexbuf }
+  | '\\' 'u' '{'
+        hex_digit hex_digit? hex_digit? hex_digit? hex_digit? hex_digit? '}'
```

Comment:
What about `hex_digit+` and an explicit error message if more than 6 digits?  Otherwise the lexer falls in the generic "Illegal backslash" warning.

You might also want to allow '_' as for other integer literals (of non-fixed length).